docs/contracts.rst

```diff
@@ -908,6 +908,33 @@ Such contracts cannot be compiled (even if they contain implemented functions al
 
 If a contract inherits from an abstract contract and does not implement all non-implemented functions by overriding, it will itself be abstract.
 
+.. index:: ! contract;interface, ! interface contract
+
+**********
+Interfaces
+**********
+
+Interfaces are similar to abstract contracts, but they cannot have any functions implemented. There are further restrictions:
+
+#. Cannot inherit other contracts or interfaces.
+#. Cannot define variables.
+#. Cannot define structs.
+
+(Structs might be supported in the future if the ABI adds support to them.)
+
+Interfaces are basically limited to what the Contract ABI can represent and the conversion between the ABI and
+an Interface should be possible without any information loss.
```

Comment:
The external interface of a contract is typed in a much stricter way than the ABI and I think interface contracts benefit from that.